pkg/template/template.go

```diff
@@ -15,7 +15,11 @@ import (
 	"github.com/openshift/origin/pkg/util/stringreplace"
 )
 
-var parameterExp = regexp.MustCompile(`\$\{([a-zA-Z0-9\_]+)\}`)
+// match ${KEY}, KEY will be grouped
+var stringParameterExp = regexp.MustCompile(`\$\{([a-zA-Z0-9\_]+?)\}`)
+
+// match ${{KEY}} exact match only, KEY will be grouped
+var nonStringParameterExp = regexp.MustCompile(`^\$\{\{([a-zA-Z0-9\_]+)\}\}$`)
```

Comment:
i think we can't safely error because of backward compatibility.  Someone could legitimately have "${{BLAH}}" in a template somewhere. (or "${{BLA H}}" or "B${{LAH}}" or any of the other permutations the regex currently ignores).  (Arguably even introducing the ${{KEY}} syntax could break existing templates that for some reason contained a value ${{KEY}} reference in a string field, but presumably we have to accept that.  So perhaps you want to make the argument that since we're already introducing a (slightly) breaking change, we can expand the number of scenarios under which people will be broken?)

That said, you'll ultimately get an error anyway if you were attempting parameter substitution into a non-string field and the substitution didn't happen because you didn't provide a valid parameter name, or because you included a prefix/suffix on the ${{KEY}}} parameter reference.

that is, you'll ultimately get an error because the resulting api object won't be valid (will contain a string value for an int/boolean/float field)